src/Dynatrace.OpenTelemetry.Exporter.Metrics/OneAgentMetadataEnricher.cs

```diff
@@ -0,0 +1,90 @@
+// <copyright company="Dynatrace LLC">
+// Copyright 2020 Dynatrace LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Dynatrace.OpenTelemetry.Exporter.Metrics
+{
+    /// <summary>
+    /// Queries the OneAgent to get metadata about the current process and enriches metric labels with them.
+    /// </summary>
+    public class OneAgentMetadataEnricher
+    {
+        private readonly ILogger<DynatraceMetricsExporter> _logger;
+        private readonly IFileReader _fileReader;
+
+        public OneAgentMetadataEnricher(ILogger<DynatraceMetricsExporter> logger) : this(logger, new DefaultFileReader())
+        {
+            this._logger = logger;
+        }
+
+        // Allows mocking of File.ReadAllText and File.ReadAllLines methods. When using the public constructor,
+        // the used FileReader passes the calls through to the System.IO methods.
+        internal OneAgentMetadataEnricher(ILogger<DynatraceMetricsExporter> logger, IFileReader fileReader)
+        {
+            this._logger = logger;
+            this._fileReader = fileReader;
+        }
+
+        public void EnrichWithDynatraceMetadata(ICollection<KeyValuePair<string, string>> labels)
+        {
+            var metadata = ReadOneAgentMetadata(GetMetadataFileContent());
+            foreach (var md in metadata)
+            {
+                labels.Add(new KeyValuePair<string, string>(md.Key, md.Value));
+            }
+        }
+
+        public IEnumerable<KeyValuePair<string, string>> ReadOneAgentMetadata(string[] lines)
```

Comment:
I think "ProcessMetadata" would be more precise since it already gets the lines read from the file earlier.